lib/withData.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react"
+import initEnvironment from "./createEnvironment"
+import { fetchQuery } from "react-relay"
+import RelayProvider from "./RelayProvider"
+import cookie from "cookie"
+
+export default (ComposedComponent: any, options: { query?: any } = {}) => {
+  return class WithData extends React.Component {
+    static displayName = `WithData(${ComposedComponent.displayName})`
+
+    static async getInitialProps(ctx: any) {
```

Comment:
This is the magic function, it basically receives and error, the request and the response from the HTTP request. Which means we can pull out the cookie and pass it into the environment